### PR TITLE
[Feature] Add Zero-shot Knowledge Transfer via Adversarial Belief Matching

### PR DESCRIPTION
**Pull request by wilxy**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add a data-free distillation algorithm named Zero-shot Knowledge Transfer via Adversarial Belief Matching.

## Modification

1.Add generaor (zskt_generator) and loss (at_loss) for zskt.
2.Add config, readme, image for zskt.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/distill/mmcls/zskt/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+```latex
+@article{micaelli2019zero,
+  title={Zero-shot knowledge transfer via adversarial belief matching},
+  author={Micaelli, Paul and Storkey, Amos J},
+  journal={Advances in Neural Information Processing Systems},
+  volume={32},
+  year={2019}
+}
+```
```
Comment (FreakieHuang) [inline configs/distill/mmcls/zskt/README.md:37]: add Acknowledgement: appreciate Davidgzx's contribution

File: mmrazor/models/architectures/generators/zskt_generator.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                3,
+                stride=1,
+                padding=1), nn.BatchNorm2d(self.hidden_channels // 2),
+            nn.LeakyReLU(leaky_slope, inplace=True),
+            nn.Conv2d(self.hidden_channels // 2, 3, 3, stride=1, padding=1),
+            nn.BatchNorm2d(3, affine=True))
+
+    def forward(self,
+                data: Optional[torch.Tensor] = None,
+                batch_size: int = 0) -> torch.Tensor:
```
Comment (FreakieHuang) [inline mmrazor/models/architectures/generators/zskt_generator.py:83]: batch_size defaulted to 0?

**Review by FreakieHuang — COMMENTED** [review]
Looks a-okay to me. 

add acknowledgement as comment, plz

File: configs/distill/mmcls/zskt/README.md (hunk 1 above)
Comment (wilxy) [reply configs/distill/mmcls/zskt/README.md:37]: Done

File: mmrazor/models/architectures/generators/zskt_generator.py (hunk 2 above)
Comment (wilxy) [reply mmrazor/models/architectures/generators/zskt_generator.py:83]: Revise it to 1.

File: mmrazor/models/losses/at_loss.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        super().__init__()
+        self.loss_weight = loss_weight
+
+    def forward(self, s_feature: torch.Tensor,
+                t_feature: torch.Tensor) -> torch.Tensor:
+        """"Forward function for ATLoss."""
+        loss = (self.at(s_feature) - self.at(t_feature)).pow(2).mean()
+        return self.loss_weight * loss
+
+    def at(self, x: torch.Tensor) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (wilxy) [reply mmrazor/models/losses/at_loss.py]: Done.
